cloud/defaults.go

```diff
@@ -133,19 +133,12 @@ func GenerateOSDiskName(machineName string) string {
 
 // GetDefaultImageSKUID gets the SKU ID of the image to use for the provided version of Kubernetes.
 func getDefaultImageSKUID(k8sVersion string) string {
-	if strings.HasPrefix(k8sVersion, "v1.16") {
-		return "k8s-1dot16-ubuntu-1804"
+	k8sVersion = strings.TrimPrefix(k8sVersion, "v")
+	parts := strings.SplitN(k8sVersion, ".", 3)
+	if len(parts) != 3 {
+		return ""
```

Comment:
why not just return an error?

Reply:
changed to return an error, I was on the edge about doing that but figured it was easier to keep the returns as-is, thanks for the nudge.